flang/lib/Lower/HostAssociations.cpp

```diff
@@ -73,13 +73,11 @@ static void bindCapturedSymbol(const Fortran::semantics::Symbol &sym,
                                fir::ExtendedValue val,
                                Fortran::lower::AbstractConverter &converter,
                                Fortran::lower::SymMap &symMap) {
-  if (converter.getLoweringOptions().getLowerToHighLevelFIR()) {
-    // TODO: add an indication that this is a host variable in the declare to
-    // allow alias analysis to detect this case.
+  converter.addHostAssocSymbol(&sym);
+  if (converter.getLoweringOptions().getLowerToHighLevelFIR())
     Fortran::lower::genDeclareSymbol(converter, symMap, sym, val);
```

Comment:
Have you consider adding and passing a boolean flag to genDeclareSymbol instead of using a denseset state in the bridge?
Maybe this could be a `fir::FortranVariableFlagsEnum extraFlags = fir::FortranVariableFlagsEnum::None` argument to be more generic.

Reply:
I just wanted to keep all the attributes computation in `translateSymbolAttributes`, but I can achieve the same by threading the extra flags to it.  I will do it.  Thank you!